element_data_loader/suite2p_trigger.py

```diff
@@ -0,0 +1,174 @@
+                        'two_step_registration',
+                        'roidetect',
+                        'spikedetect', 'delete_bin',
+                        'xblock', 'yblock', 'xrange', 'yrange', 'nblocks',
+                        'nframes']
+        registration_ops = {key: registration_ops[key] for key in desired_keys}
+
+    else:
+
+        print("------------RUNNING RIGID REGISTRATION------------")
+
+        registration_ops = run_s2p(ops,db)
+        desired_keys = ['xoff', 'yoff', 'do_registration', 'two_step_registration',
+                        'roidetect', 'spikedetect', 'delete_bin']
+        registration_ops = {key: registration_ops[key] for key in desired_keys}
+
+    return registration_ops
+
+
+def segmentation_suite2p(registration_ops, db):
+    """Performs cell detection using Suite2p package, 
+    Make sure the 'roidetect' key is set to True to perform cell/ roi detection
+
+    Args:
+        registration_ops (dict): reg_ops dictionary can be obtained from
+                                 registration step (CaImAn or Suite2p).
+                                 Must contain x and y shifts along with 
+                                 'do_registration'=0,
+                                 'two_step_registration'=False,
+                                 'roidetect'=True and 'spikedetect'= False
+        db (dict): dictionary that includes paths pointing towards the input 
+                   data, and path to store outputs
+
+    Returns:
+        segmentation_ops (dict): The ops dictionary is updated with additional
+                                 keys that are added 
+        data.bin: Creates and saves a binary file on your local path, if the one
+                  created during registration is deleted. If delete_bin is set 
+                  to True, the binary file is deleted after processing.
+        ops.npy: Updates the options dictionary in a file created during the 
+                 registration step and saves as a numpy array file on the 
+                 specified path.
+        F.npy: Stores an array of fluorescence traces
+        Fneu.npy: Returns an array of neuropil fluorescence traces
+        iscell.npy: Specifies whether an ROI is a cell
+        stat.npy: Returns a list of statistics computed for each cell
+        spks.npy: Returns an empty file, it is updated during the deconvolution
+                  step with deconvolved traces
+    """
+
+    if registration_ops['do_registration'] or not (registration_ops['roidetect']) or registration_ops['spikedetect']:
+        warnings.warn('You are running segmentation with Suite2p. Requirements\
+                     include do_registration=0, roidetect=True, \
+                     spikedetect=False. The ops dictionary has differing\
+                     values. The flags will be set to the required values.')
+    registration_ops.update(do_registration = 0,
+                    roidetect = True,
+                    spikedetect = False)
```

Comment:
```suggestion
    if registration_ops['do_registration'] or not (registration_ops['roidetect']) or registration_ops['spikedetect']:
        warnings.warn('You are running segmentation with Suite2p. Requirements\
                     include do_registration=0, roidetect=True, \
                     spikedetect=False. The ops dictionary has differing\
                     values. The flags will be set to the required values.')
         registration_ops.update(do_registration = 0,
                    roidetect = True,
                    spikedetect = False)
```

Reply:
Thank you for the suggestions, @kabilar. I kept the rest of the code outside the 'if' statement because incase the user forgets to change the flag, then the code will not run at all (this will be against what we have in the warning : we mentioned that the flags will be automatically updated even if the user sets them incorrectly). Please correct me if I misunderstood your suggestion here. Thank you!